cluster-controller/src/main/java/io/strimzi/controller/cluster/resources/AbstractCluster.java

```diff
@@ -577,4 +578,12 @@ protected ConfigMap patchConfigMap(ConfigMap cm, Map<String, String> data) {
     protected static EnvVar buildEnvVar(String name, String value) {
         return new EnvVarBuilder().withName(name).withValue(value).build();
     }
+
+    /**
+     * Gets the given container's environment.
+     */
+    protected static Map<String, String> containerEnv(Container container) {
```

Comment:
I think we should agree on a single form about writing Javadoc. In general I always add @param and @return even if the names are understandable. I am not saying that my form is the right one but we should agree on having only one "unambiguous" format.
In any case I would call this method `containerEnvVars`.

Reply:
I agree we should try to be more consistent with javadoc. Since this isn't a library, I feel that using `@param`, `@return` etc can take a lot of space without really adding much. In this particular instance they would add no value beyond what's in the description. I would end up with a method which was 3 lines of code, and 5 lines of doc. And that's doc of an internal method which we're expect no one to call except for strimzi developers. That just feels like pointless bloat to me.

That said, I'm happy to go with adding `@` tags to doc if that's the consensus.

How about `containerEnvironment()`? `containerEnvVars()` doesn't convey to me that it's the whole environment (names and values) that's returned, rather than just names.

Reply:
Why does `EnvVars` suggest to you that it doesn't return env vars as a whole so names + values but just names ? :-)

Reply:
Yeah, it's a subtle distinction (and maybe one that only exists in my mind!), but talking about a process' environment conveys to me that it's the whole thing, but the plural "environment variables" is just a bit ambiguous about whether it's just the keys or the keys+values.

In fact, the `container` part of the name is also a bit unnecessary, since we're passing a container in it's fairly obvious it must be the container's environment.

Reply:
But a `variable` is made by a name and a value (at least in in my mind). It's not related to the fact that they are part of the environment where the application is running.
To me process environment means different things ... not just the env vars (or not env vars at all) but more something related to resources (CPU, memory, ..) and JVM ...

Reply:
TBH I have to agree with @ppatierno here. I was thinking whether I should raise it as well, but it didn't seem important enough for me. But that was exactly the thought I had ... _Container Environment_ is for me a lot more than just the environment variables.

Reply:
@scholzj It wasn't important for me as well ... I was giving up and let @tombentley using that name :P